host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java

```diff
@@ -285,9 +307,9 @@ protected HostRegistrationStepHandler(final TransformerRegistry transformerRegis
 
         @Override
         public void execute(final OperationContext context, final ModelNode operation) throws OperationFailedException {
-            // First lock the domain controller
-            context.acquireControllerLock();
-            // Check with the controller lock held
+            // acquires the shared-mode read lock.
+            operationExecutor.beginRegistration(registrationContext.activeOperation.getOperationId(), registrationContext.hostInfo);
+
```

Comment:
Re: exception handling, if beginRegistration throws InterruptedException, the following is equivalent logic to what happens when context.acquireControllerLock() is called.

```
try {
    operationExecutor.beginRegistration(registrationContext.activeOperation.getOperationId(), registrationContext.hostInfo);
} catch (InterruptedException e) {
    Thread.currentThread().interrupt();
    throw ControllerLogger.ROOT_LOGGER.operationCancelledAsynchronously();
}
```

If an InterruptedException occurs in an OperationStepHandler, Thread.currentThread().interrupt() should be called so when OSH.execute() returns the OperationContext is aware of the request to stop execution and its logic for dealing with that gets a chance to run. So the handling of InterruptedException can happen here, or in the deeper layers, but whoever catches it needs to call Thread.currentThread().interrupt().